src/NUnitFramework/framework/Assert.cs

```diff
@@ -270,16 +272,95 @@ public static void Contains(object expected, ICollection actual)
 
         #region Multiple
 
-        ///// <summary>
-        ///// If an assert fails within this block, execution will continue and 
-        ///// the errors will be reported at the end of the block.
-        ///// </summary>
-        ///// <param name="del">The test delegate</param>
-        //public static void Multiple(TestDelegate del)
-        //{
-        //    del();
-        //}
+        /// <summary>
+        /// Wraps code containing a series of assertions, which should all
+        /// be executed, even if they fail. Failed results are saved and
+        /// reported at the end of the code block.
+        /// </summary>
+        /// <param name="testDelegate">A TestDelegate to be executed in Multiple Assertion mode.</param>
+        public static void Multiple(TestDelegate testDelegate)
+        {
+            TestExecutionContext context = TestExecutionContext.CurrentContext;
+            Guard.OperationValid(context != null, "Assert.Multiple called outside of a valid TestExecutionContext");
+
+            context.MultipleAssertLevel++;
+
+            try
+            {
+                testDelegate();
+            }
+            finally
+            {
+                context.MultipleAssertLevel--;
+            }
+
+            if (context.MultipleAssertLevel == 0)
+            {
+                int count = context.CurrentResult.AssertionResults.Count;
+
+                if (count > 0)
+                {
+                    var writer = new TextMessageWriter("Multiple Assert block had {0} failure(s).", count);
+                    writer.WriteLine();
+
+                    int counter = 0;
+                    foreach (var assertion in context.CurrentResult.AssertionResults)
+                    {
+                        writer.WriteLine(string.Format("  {0}) {1}", ++counter, assertion.Message));
```

Comment:
Sorry, I should have been more explicit. The indent is fine, I was thinking you need the opening parenthesis.